src/main/java/uk/gov/hmcts/reform/prl/services/FeeService.java

```diff
@@ -211,18 +211,18 @@ private static Optional<FeeType> fromOtherPartyConsentAndHearing(String otherPar
 
     private static FeeType fromOtherPartyConsentAndNotice(String otherPartyConsent, String notice) {
 
-        if (otherPartyConsent.equals(YES)) {
+        if (isNotBlank(otherPartyConsent) && otherPartyConsent.equals(YES)) {
             return C2_WITHOUT_NOTICE;
         } else {
-            if (notice.equals(NO)) {
-                return C2_WITHOUT_NOTICE;
-            } else if (notice.equals(YES)) {
-                return C2_WITH_NOTICE;
-            } else {
-                return null;
+            if (isNotBlank(notice)) {
+                if (notice.equals(NO)) {
+                    return C2_WITHOUT_NOTICE;
+                } else if (notice.equals(YES)) {
+                    return C2_WITH_NOTICE;
+                }
```

Comment:
Always compare with the Constant, then you can avoid null pointer exception as well as an extra if block.
So it should be - if (YES.equals(otherPartyConsent)) and so on.

Reply:
Done